tests/Unit/Evolution/Systems/NewtonianEuler/BoundaryCorrections/Test_Hll.cpp

```diff
@@ -0,0 +1,174 @@
+#include "Evolution/Systems/NewtonianEuler/BoundaryCorrections/Factory.hpp"
+#include "Evolution/Systems/NewtonianEuler/BoundaryCorrections/Hll.hpp"
+#include "Evolution/Systems/NewtonianEuler/System.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp"
+#include "Helpers/Evolution/DiscontinuousGalerkin/Range.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+struct ConvertPolytropic {
+  using unpacked_container = bool;
+  using packed_container = EquationsOfState::PolytropicFluid<false>;
+  using packed_type = bool;
+
+  static inline unpacked_container unpack(
+      const packed_container& /*packed*/,
+      const size_t /*grid_point_index*/) noexcept {
+    return true;
+  }
+
+  [[noreturn]] static inline void pack(
+      const gsl::not_null<packed_container*> /*packed*/,
+      const unpacked_container& /*unpacked*/,
+      const size_t /*grid_point_index*/) {
+    ERROR("Should not be converting an EOS from an unpacked to a packed type");
+  }
+
+  static inline size_t get_size(const packed_container& /*packed*/) noexcept {
+    return 1;
+  }
+};
+
+struct ConvertIdeal {
+  using unpacked_container = bool;
+  using packed_container = EquationsOfState::IdealFluid<false>;
+  using packed_type = bool;
+
+  static inline unpacked_container unpack(
+      const packed_container& /*packed*/,
+      const size_t /*grid_point_index*/) noexcept {
+    return false;
+  }
+
+  [[noreturn]] static inline void pack(
+      const gsl::not_null<packed_container*> /*packed*/,
+      const unpacked_container& /*unpacked*/,
+      const size_t /*grid_point_index*/) {
+    ERROR("Should not be converting an EOS from an unpacked to a packed type");
+  }
+
+  static inline size_t get_size(const packed_container& /*packed*/) noexcept {
+    return 1;
+  }
+};
+
+struct DummyInitialData {
+  using argument_tags = tmpl::list<>;
+  struct source_term_type {
+    using sourced_variables = tmpl::list<>;
+    using argument_tags = tmpl::list<>;
+  };
+};
+
+namespace helpers = TestHelpers::evolution::dg;
+
+template <size_t Dim, typename EosType>
+void test(const gsl::not_null<std::mt19937*> gen, const size_t num_pts,
+          const EosType& equation_of_state) {
+  tuples::TaggedTuple<hydro::Tags::EquationOfState<EosType>> volume_data{
+      equation_of_state};
+  tuples::TaggedTuple<
+      helpers::Tags::Range<NewtonianEuler::Tags::MassDensityCons>,
+      helpers::Tags::Range<
+          NewtonianEuler::Tags::SpecificInternalEnergy<DataVector>>>
+      ranges_for_conservation{std::array<double, 2>{{1.0e-30, 1.0}},
```

Comment:
[optional, more an FYI] with c++17 you can write `std::array{1.0e-30, 1.0}` and it will figure out this is a `std::array<double, 2>`